llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -14952,6 +14952,30 @@ const SCEVAddRecExpr *ScalarEvolution::convertSCEVToAddRecWithPredicates(
   if (!AddRec)
     return nullptr;
 
+  // Check if any of the transformed predicates is known to be false. In that
+  // case, it doesn't make sense to convert to an predicated AddRec, as the
+  // versioned loop will never execute.
+  for (const SCEVPredicate *Pred : TransformPreds) {
+    auto *WrapPred = dyn_cast<SCEVWrapPredicate>(Pred);
+    if (!WrapPred || WrapPred->getFlags() != SCEVWrapPredicate::IncrementNSSW)
+      continue;
+
+    const SCEVAddRecExpr *AddRecToCheck = WrapPred->getExpr();
```

Comment:
Do we need to check that the addrecs are on the same loop? Or is that already guaranteed?

Reply:
I think for all cases the loop should match in practice, but I updated to use the loop from AddRecToCheck, thanks